app/src/main/java/io/homeassistant/companion/android/notifications/MessagingManager.kt

```diff
@@ -1935,6 +1943,29 @@ class MessagingManager @Inject constructor(
         }
     }
 
+    private fun setAppLock(data: Map<String, String>) {
+        try {
+            if (data.containsKey("app_lock_enabled")) {
+                runBlocking {
+                    authenticationUseCase.setLockEnabled(data["app_lock_enabled"]!!.toBooleanStrict())
+                }
+            }
+            if (data.containsKey("app_lock_timeout")) {
+                runBlocking {
+                    integrationUseCase.sessionTimeOut(data["app_lock_timeout"]!!.toInt())
```

Comment:
Should we do some validation here like ensuring we get a positive number instead of negative? Maybe also use `toIntOrNull()` so we can default to current setting if there is a `null` value?

Reply:
So it looks like we are accepting negative values here, I think we should not allow values that are not valid. As of now it will save any negative value. The frontend does not allow a user to enter a negative value but only a positive whole integer.